### PR TITLE
feat(broker): support error end events

### LINKED ISSUE #3510: Throw an error from an end event
**Is your feature request related to a problem? Please describe.**
I want to throw an error inside my workflow to indicate a business failure (i.e. non-technical). The error should be caught by a boundary event or an event subprocess.

**Describe the solution you'd like**
* an error can be thrown using an error end event
* the end event references an error with an `errorCode` (any string)

**Describe alternatives you've considered**
none

**Additional context**

requires #3497 
requires #3498

### PR DESCRIPTION
**Pull request by saig0**
## Description

* validate and transform error end events
* throw an error event when the end event is activated
* create an incident if the error is not caught
* resolving the incident will create a new one
* extract logic to handle the error events

## Related issues

closes #3510 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Comment by saig0** [comment]
SonarCloud complains about duplicated code in the model API (`AbstractEndEventBuilder`) :roll_eyes:

File: engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/endevent/ErrorEventHandle.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+  private static final DirectBuffer NO_VARIABLES = new UnsafeBuffer();
+
+  private final CatchEventTuple catchEventTuple = new CatchEventTuple();
+
+  private final WorkflowState workflowState;
+  private final ElementInstanceState elementInstanceState;
+  private final EventHandle eventHandle;
+
+  public ErrorEventHandle(final WorkflowState workflowState, final KeyGenerator keyGenerator) {
```
Comment (MiguelPires) [inline engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/endevent/ErrorEventHandle.java]: Why is this a `Handle` instead of `Handler`?

**Review by miguelpires — APPROVED** [review]
Looks good. Pointed out two minor things but they're not blockers

**Comment by saig0** [comment]
@MiguelPires, thank you for the review. I applied your hints.

**Pull request merged**